structuralcodes/sections/_generic.py

```diff
@@ -1047,6 +1046,9 @@ def _np_space(a, b, n, type, endpoint):
             raise ValueError(f'Type of spacing not known: {type}')
 
         # For generation of fields 1 and 2 pivot on positive strain
+        field_num = np.repeat(
+            [1, 2, 3, 4, 5, 6], [num_1, num_2, num_3, num_4, num_5, num_6]
```

Comment:
This more a question than a request for change: would it be more slightly more Pythonic if we zero-index the fields?

Reply:
I totally see your concern. I thought a lot. I decided to use 1-6 because it is what you find in books... If you prefer we can change it to 0-5...

Reply:
Then I suggest we go for 1-6 as in the books 😊